### ISSUE TITLE
How to set a old Linux Kernel and EABI4

### ISSUE THREAD

**Issue by nyckmaia** [crosstool-ng/crosstool-ng#1542]
My crosstool-ng project builds good with no errors.
My problem is: To match the final binary that runs inside my target, I need to change 2 Crosstool-ng configurations.
They are the `EABI` and `Linux Version`.

When I build my application using my Crosstool-ng toolchain, the output binary is:
```
ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.3, for GNU/Linux 3.2.101, with debug_info, not stripped
```
**BUT** my target needs a final binary little bit different:
```
ELF 32-bit LSB executable, ARM, EABI4 version 1 (SYSV), dynamically linked, interpreter /lib/ld-linux.so.3, for GNU/Linux 2.6.33, with debug_info, not stripped
```
## The diferences are:
- Must change from `EABI5` to `EABI4`
- Must change from `/lib/ld-linux.so.3, for GNU/Linux 3.2.101` to `/lib/ld-linux.so.3, for GNU/Linux 2.6.33`

I already tried to use the `./ct-ng menuconfig` to set the desired `2.6.33` kernel, but the oldest in the list is the `3.2.101`.

**How can I change these 2 options?**

**CrossTool-Ng version:** latest (cloned from master today 28 June 2021)
**Host OS:** Ubuntu 20.04 LTS x64
**Target/sample used:** arm-unknown-linux-gnueabi

Here is my `.config` file.

[ct_config.txt](https://github.com/crosstool-ng/crosstool-ng/files/6726872/ct_config.txt)

**Comment by cpackham** [comment]
You might get some mileage out of using one of the `-mabi` options in your `CFLAGS`. I'm not actually sure what they correspond to but [this](http://kanj.github.io/elfs/book/armMusl/cross-tools/abi.html) suggests `apcs-gnu` might be what you want.

To get at the older kernel version you need to select `OBSOLETE` and a sufficiently old glibc (older than 2.24). I will warn that those kinds of options get little to no testing so you may have to play around to get a configuration that will successfully build.

**Comment by nyckmaia** [comment]
Now I saw the Linux kernel `2.6.32` in the `./ct-ng menuconfig`. So the kernel version is solved to me.

The only thing that I need to change yet is move my output application binary from `EABI5` to `EABI4`.

Seaching on internet I got 2 different configurations about `EABI`.
On [this page](http://kanj.github.io/elfs/book/armMusl/cross-tools/abi.html) the `eabi` version can be configured using:
```
-meabi=aapcs-linux // Recommended for EABI. Standard Linux 32 bit (int) enums
or
-meabi=apcs-gnu // Recommended for OABI. The OABI (old ABI) is called apcs-gnu and supports ARMv4 and older cores. Generally the OABI is not used by modern ARM processors.
```

But, in [this page](https://sourceware.org/binutils/docs/as/ARM-Options.html#ARM-Options) I got that `eabi` version can be configured using:

```
-meabi=gnu
or
-meabi=4
or
-meabi=5
This option specifies which EABI version the produced object files should conform to. 
```

My target is a `armv6`, so I think that this must use `EABI` and not `OABI` like the first link above, right?

So, in the crosstool-ng configurations I tried to set the both possibilities `-meabi=4` and `-meabi=apcs-gnu`, but I got a compile error.

How can I set the crosstool-ng to enable the compiled `gcc` to build binaries that uses `EABI version 4`?
